pkg/action/util/util.go

```diff
@@ -318,3 +325,123 @@ func GetPodGrandInfo(kclient *client.Clientset, pod *api.Pod) (string, string, e
+	}
+}
+
+// Gets status of replicas and ready replicas of a controller.
+// Currently, it supports replication controllers, replica sets, and deployments
+func GetReplicaStatus(kubeClient *client.Clientset, namespace, contName, contKind string) (int, int, error) {
+	switch contKind {
+	case util.KindReplicationController:
+		if cont, err := kubeClient.CoreV1().ReplicationControllers(namespace).Get(contName, metav1.GetOptions{}); err != nil {
+			return 0, 0, err
+		} else {
+			return int(cont.Status.Replicas), int(cont.Status.ReadyReplicas), nil
+		}
+	case util.KindReplicaSet:
+		if cont, err := kubeClient.ExtensionsV1beta1().ReplicaSets(namespace).Get(contName, metav1.GetOptions{}); err != nil {
+			return 0, 0, err
+		} else {
+			return int(cont.Status.Replicas), int(cont.Status.ReadyReplicas), nil
+		}
+	case util.KindDeployment:
+		if cont, err := kubeClient.ExtensionsV1beta1().Deployments(namespace).Get(contName, metav1.GetOptions{}); err != nil {
+			return 0, 0, err
+		} else {
+			return int(cont.Status.Replicas), int(cont.Status.ReadyReplicas), nil
+		}
+	default:
+		return 0, 0, fmt.Errorf("unsupported kind: %s", contKind)
+	}
+}
+
+func getSelectorFromController(kubeClient *client.Clientset, namespace, contName, contKind string) (map[string]string, error) {
+	switch contKind {
+	case util.KindReplicationController:
+		if cont, err := kubeClient.CoreV1().ReplicationControllers(namespace).Get(contName, metav1.GetOptions{}); err != nil {
+			return nil, err
+		} else {
+			return cont.Spec.Selector, nil
+		}
+	case util.KindReplicaSet:
+		if cont, err := kubeClient.ExtensionsV1beta1().ReplicaSets(namespace).Get(contName, metav1.GetOptions{}); err != nil {
+			return nil, err
+		} else {
+			return cont.Spec.Selector.MatchLabels, nil
+		}
+	case util.KindDeployment:
+		if cont, err := kubeClient.ExtensionsV1beta1().Deployments(namespace).Get(contName, metav1.GetOptions{}); err != nil {
+			return nil, err
+		} else {
+			return cont.Spec.Selector.MatchLabels, nil
+		}
+	default:
+		return nil, fmt.Errorf("unsupported kind: %s", contKind)
+	}
+}
+
+// Finds all child pods of a controller.
+// Currently, it supports replication controllers, replica sets, and deployments
+func FindPodsByController(kubeClient *client.Clientset, namespace, contName, contKind string) ([]*api.Pod, error) {
+	selector, err := getSelectorFromController(kubeClient, namespace, contName, contKind)
+	if err != nil {
+		return nil, err
+	}
+
+	// Finds all pods with the labels of the controller
+	podList, err := kubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{
+		LabelSelector: labels.Set(selector).AsSelector().String(),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error getting all pods by %s %s/%s: %s", contKind, namespace, contName, err)
+	}
+
+	pods := []*api.Pod{}
+
+	for _, pod := range podList.Items {
+		_, name, err := GetPodParentInfo(&pod)
+		if err != nil {
+			continue
+		}
+		if name == contName {
+			pod := pod
```

Comment:
this is confusing `pod := pod`. Alternatively, traverse the podList by index:
```go
 for i := range podList.Items {
    pod := &(podList.Items[i])

    ...
   pods = append(pods, pod)
}
```

Reply:
fixed